### PR TITLE
get_dog(id) function created in resources

### LINKED ISSUE #24: Create internal function and api get information on a dog
Related to #16 

We want to make a **get** request to: `/rsrc/dog/<dog id>` and get back info on the dog. Specifically the name. So the response should look something like:
```json
{
  "id": 10,
  "name": "REX"
}
```
We want the option to update this with more information in the future.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: server/registry/resources.py [hunk 1]
```diff
@@ -41,6 +41,11 @@ def get_dogs(self):
 	def add_dog(self):
 		self.all_dogs.append(self)
 
+    def get_dog(id1, self):
+		for dog in self.all_dogs:
+			if dog.id == id1:
+				return {"name":dog.name,"id":dog.id}
```
Comment (cwlowder) [inline server/registry/resources.py]: You should return the actual dog object here. And then create a web api for getting the actual dog as this json

**Review by cwlowder — CHANGES_REQUESTED** [review]

File: server/registry/resources.py [hunk 2]
```diff
@@ -41,6 +41,11 @@ def get_dogs(self):
 	def add_dog(self):
 		self.all_dogs.append(self)
 
+    def get_dog(id1, self):
```
Comment (chenfeiyu132) [inline server/registry/resources.py]: why is this called id1?

**Review by chenfeiyu132 — COMMENTED** [review]
Take a look at comments

File: server/registry/resources.py (hunk 2 above)
Comment (ghost) [reply server/registry/resources.py]: I wasnt sure if there will be any errors cause an id has already been defined so just called it id1

File: server/registry/resources.py [hunk 3]
```diff
@@ -38,6 +38,11 @@ def available(self, resource):
 	def get_dogs(self):
 		return self.all_dogs
 
+	def get_dog_object(self, id1):
```
Comment (cwlowder) [inline server/registry/resources.py]: change id1 to dog_id

File: server/app.py [hunk 4]
```diff
@@ -198,6 +198,12 @@
     return json({"id": id})
 
 
+@app.route("/rsrc/dog/<dog_id>", methods=["GET",])
+async def get_dog_json(request: Request, dog_id: int):
+    dog = registry.get_dog_object(dog_id)
+    return json({"id":dog.id, "name":dog.name})
+
+
 @app.websocket("/rsrc/dog/<dog_id>")
```
Comment (cwlowder) [inline server/app.py]: Can you rename this to `@app.websocket(/ws/dog/<dog_id>)`?

**Review by cwlowder — CHANGES_REQUESTED** [review]

**Pull request closed without merging**